wgsl/index.bs

```diff
@@ -6538,21 +6536,21 @@ A <dfn noexport>decrement statement</dfn> subtracts 1 from the contents of a var
     | [=syntax/lhs_expression=] [=syntax/minus_minus=]
 </div>
 
-The expression [=shader-creation error|must=] evaluate to a reference with an [=integer scalar=] [=store type=] and [=access/read_write=] [=access mode=].
+The expression [=shader-creation error|must=] evaluate to a reference with a [=type/concrete=] [=integer scalar=] [=store type=] and [=access/read_write=] [=access mode=].
 
 <table class='data'>
   <thead>
     <tr><th>Precondition<th>Statement<th>Description
   </thead>
   <tr algorithm="increment statement">
     <td class="nowrap">|r| : ref&lt;|AS|,|T|,[=access/read_write=]&gt;,<br>
-        |T| is [=integer scalar=]<br>
+        |T| is a [=type/concrete=] [=integer scalar=]<br>
     <td class="nowrap">|r|`++`
     <td>Adds 1 to the contents of memory referenced by |r|.
         <br>Same as |r| += |T|(1)
   <tr algorithm="decrement statement">
     <td class="nowrap">|r| : ref&lt;|AS|,|T|,[=access/read_write=]&gt;,<br>
-        |T| is [=integer scalar=]<br>
+        |T| is a [=type/concrete=] [=integer scalar=]<br>
```

Comment:
(Further down).

Must also fix the type rule precondition for switch.

Reply:
Should also fix  in var declaration rule about zero-initialization where it says:

> If the [store type](https://gpuweb.github.io/gpuweb/wgsl/#store-type) is an [atomic type](https://gpuweb.github.io/gpuweb/wgsl/#atomic-type), the [zero value](https://gpuweb.github.io/gpuweb/wgsl/#zero-value) is that of the underlying type ([integer scalar](https://gpuweb.github.io/gpuweb/wgsl/#integer-scalar)).

That parenthetical integer-scalar should be concrete integer scalar.